lk2nd/smb1360/smb1360-idol347.c

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * Detect batt_id_uv and set qcom,otp-rslow-config accordingly
+ */
+
+#include <debug.h>
+#include <pm8x41.h>
+#include <pm8x41_adc.h>
+#include "smb1360.h"
+
+static const struct smb1360_battery idol347_batteries[] = {
+        [0] = {
+		.name = "0 BattID=606000",
+		.rslow_config = (uint8_t[]) { 0x71, 0x73, 0x29, 0x78 },
+        },
+        [1] = {
+		.name = "1 BattID=32000",
+		.rslow_config = (uint8_t[]) { 0xDA, 0x71, 0x34, 0x71 },
+        },
+};
```

Comment:
The array here seems pointless, just create two separate structs for this like

```c
static const struct smb1360_battery battery_tlp020k2 = {
	.name = "TLP020K2",
	.rslow_config = (uint8_t[]) { 0x71, 0x73, 0x29, 0x78 },
};

static const struct smb1360_battery battery_tlk020kj = {
	.name = "TLK020KJ",
	.rslow_config = (uint8_t[]) { 0xDA, 0x71, 0x34, 0x71 },
};
```

The names can be found in this comment: https://github.com/msm8916-mainline/android_kernel_qcom_msm8916/blob/e87abaceaaf3e67c25e1a734838ed277dc0c44a9/drivers/power/idol347/smb1360-charger-fg.c#L4159-L4160